UnitTest/tests/test_DeviceData_rmsg-definition.txt

```diff
@@ -1,130 +1,146 @@
 defmod test_DeviceData_rmsg UnitTest dummyDuino ( 
- {  
- 
-  	my $mock = Mock::Sub->new; 
-	use JSON; 
-  	use List::Util qw[min max];
+  {  
+    note("versionmodul: ".InternalVal($target, "versionmodul", "unknown"));
+    note("versionProtocols: ".InternalVal($target, "versionProtocols", "unknown"));
+    my $whitelist_IDs = AttrVal($target,"whitelist_IDs",undef);	# save whitelist_IDs to set back
+    note("attribute whitelist_IDs save from system");
+    my $maxMuMsgRepeat = AttrVal($target,"maxMuMsgRepeat",undef);	# save maxMuMsgRepeat to set back
+    note("attribute maxMuMsgRepeat save from system");
+    CommandAttr($targetHash,"$target maxMuMsgRepeat 10");
```

Comment:
Wieso hast Du in dem Test die maximale Anzahl an repeats erhöht?
Wäre es nicht besser wie bisher mit dem Standardwert zu testen?

Reply:
Wenn der Standardwert aktiv ist, so sehen wir mit einem Blick nicht, welche Nachrichten sehr viele Wiederholungen besitzen.

Mit welchem Wert wir testen ist gal, aber bei einem höheren Wert siehst du auf jedenfall schneller wenn etwas abhanden kommen könnte. 

Ich denke mal so "dumm", das Bsp. bei 7 Wiederholungen immer nur 4 angezeigt werden. Sollte dann im Programm mal ein Fehler sein, so können 3 Wiederholungen abhanden kommen bis wir etwas mitbekommen. Wenn ich so 7 zu 7 testen kann, sehe ich sofort wenn auf einmal etwas fehlt.

Reply:
Ich bin mir unsicher, ob wir das wirklich sehen müssen wie viele Wiederholungen da drinnen stecken, solange die Mindestanzahl erreicht wird.

Reply:
Gern können wir es auch zurück nehmen aber dann kann keiner die Frage beantworten,
"Gibt es Nachrichten die mehr als 4 oder x Wiederholungen haben?"

Derzeit geht die Frage zu beantworten, weil man in die Daten schaut und nach einem Wert > 4 schauen kann.

>Ich bin mir unsicher, ob wir das wirklich sehen müssen wie viele Wiederholungen da drinnen stecken, solange die Mindestanzahl erreicht wird.

Für was haben wir dann damals diesen Wert aufnehmen wollen in das JSON?

Reply:
Vermutlich um verifizieren zu können ob die Routinen mehrere Wiederholungen erkennt.
Wenn zwei enthalten sind und zwei werden erkannt ist es auch die maximale Anzahl die erreicht werden kann.
Bei 11 Wiederholungen wäre jetzt ja auch bei 10 Schluss. (Vermutlich gibt es den Fall mit 11 nicht oder)?

Reply:
>Bei 11 Wiederholungen wäre jetzt ja auch bei 10 Schluss. (Vermutlich gibt es den Fall mit 11 nicht oder)?

Derzeit noch nicht. Der Maximale Wert sind glaube (ohne nun nachzusehen) 7. Wir befinden uns in einem Momentanstand. Vielleicht ist es nächstes Jahr mal, das wir eine Nachricht von 9 Wiederholungen haben.

Wir können das alles subjektiv betrachten. Gern übernehme ich den Wert 4 als Standard ABER dann sehen wir auch nicht die Wiederholungsanzahl von Nachrichten wenn ein Signal mehr als 4x Wiederholt, da wir kein separates Datenfeld Repeats besitzen. Das ist ebenso ein Grund wieso ich den Wert auf 10 gesetzt habe. Somit Prüfen wir den Wert bis 10 und wenn ein Signal mal mehr als 4 Wiederholungen sendet (was mehrfach vorkommt) so sehe ich es sofort in dem Dispatch Wert (ausgeommen von MC/MS/ dort steht es im Comment) Kurzum, mit der Erhöhung auf 10 ersparen wir uns das separate Feld Repeats separat. Manchmal ist es sehr Hilfreich zu wissen, wieviel Nachrichten herauskommen müssten.

Reply:
Ok bin überzeugt, dann können wir es aber auch gleich auf 99 setzen oder? Das ist dann eindeutig, was die Absicht des hohen Wertes ist.

Reply:
Dann machen wir es so.
Ich ändere es ab.